python/ray/data/_internal/execution/streaming_executor_state.py

```diff
@@ -310,6 +314,11 @@ def select_operator_to_run(
         if state.num_queued() > 0 and _execution_allowed(op, cur_usage, limits)
     ]
 
+    # If no ops are allowed to execute due to resource constraints, try to trigger
+    # cluster scale-up.
+    if not ops:
```

Comment:
Should we guard this with a flag? Some users may not want to invoke autoscaling.

Reply:
The user can disable auto scaling at the cluster level in that case. The expectation is that workloads should always trigger auto scaling if possible.